# RNA-seq-workshop

This RNA-seq workshop aims to get you started with your own RNA-seq analysis and assumes you are already familiar with the basics of bash and R.

We will be using NeSI HPC for the analysis so make sure you have a NeSI account and you are able to login.

Go through the workshop prep files to ensure you are set to go: <a href="https://github.com/foreal17/RNA-seq-workshop/blob/master/Prep_Files/README.md">Prep_Files</a>


## Workshop outline

  * Overview of RNA-seq
  * What is a Fastq file?
  * How to QC a fastq ?
  * How to trim a sequence file?
    * Quality trimming
    * Adaptor trimming
  * How to align the sequencing reads on a reference using HISAT2 
    * creating a reference sequence
    * aligning reads on a reference
  * Defining alignment format
  * Introduction to SAMtools
  * How to get a count matrix
    * What is an annotation file
    * How to use featureCount

  * __LUNCH BREAK__

  * Differential Gene Expression analysis
  - Using DESeq2, edgeR and Limma
  * Overrepresentation analysis (Gene Ontology)
  
  * __DISCUSSION AND FOLLOW-UP__
 
--- 

  ## Attribution notice
  
- The material used to prepare for the workshop was extracted from Assoc. Prof. Michael Black lectures for Stat435 course.

- Some code used in this workshop has been extracted from a collection of presentations, workshops and lectures by Prof. Thomas Girke, Institute for Integrative Genome Biology, UC Riverside: <a href="http://girke.bioinformatics.ucr.edu/">Link_1</a> and <a href="http://faculty.ucr.edu/~tgirke/HTML_Presentations/Manuals/MCBIOS2015/Rrnaseq/Rrnaseq.pdf">Link_2</a>

---
